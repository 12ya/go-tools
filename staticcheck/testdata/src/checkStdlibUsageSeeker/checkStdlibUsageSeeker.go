package pkg

import (
	"io"
	myio "io"
	"os"
)

func fn() {
	const SeekStart = 0
	var s io.Seeker
	s.Seek(0, 0)
	s.Seek(0, io.SeekStart)
	s.Seek(io.SeekStart, 0)   //@ diag(`the first argument of io.Seeker is the offset`)
	s.Seek(myio.SeekStart, 0) //@ diag(`the first argument of io.Seeker is the offset`)
	s.Seek(SeekStart, 0)

	var f *os.File
	f.Seek(io.SeekStart, 0) //@ diag(`the first argument of io.Seeker is the offset`)
}
