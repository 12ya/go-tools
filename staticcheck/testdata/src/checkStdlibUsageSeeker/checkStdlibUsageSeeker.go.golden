package pkg

import (
	"io"
	myio "io"
	"os"
)

func fn() {
	const SeekStart = 0
	var s io.Seeker
	s.Seek(0, 0)
	s.Seek(0, io.SeekStart)
	s.Seek(0, io.SeekStart)   //@ diag(`the first argument of io.Seeker is the offset`)
	s.Seek(0, myio.SeekStart) //@ diag(`the first argument of io.Seeker is the offset`)
	s.Seek(SeekStart, 0)

	var f *os.File
	f.Seek(0, io.SeekStart) //@ diag(`the first argument of io.Seeker is the offset`)
}
